lib/src/rules/cascade_invocations.dart

```diff
@@ -96,57 +117,147 @@ class CascadeInvocations extends LintRule {
   AstVisitor getVisitor() => _visitor;
 }
 
-class _Visitor extends SimpleAstVisitor {
-  final LintRule rule;
-  _Visitor(this.rule);
+class _ExpressionBox {
```

Comment:
The name `_ExpressionBox` doesn't really convey any semantic value. It would be good if you could come up with a better name. (One approach that often works for me is to write a documentation comment because it forces me to think about what the object really represents.) Minimally, add a comment to explain how this is being used.